### PR TITLE
feat: expose reserveIds API

### PR DESCRIPTION
**Pull request by athakor**

### CODE REVIEW

File: google-cloud-datastore/src/main/java/com/google/cloud/datastore/Datastore.java [hunk 1]
```diff
@@ -168,6 +169,23 @@
    */
   List<Key> allocateId(IncompleteKey... keys);
 
+  /**
+   * Reserve a list of IDs from a complete key
```
Comment (rahulKQL) [inline google-cloud-datastore/src/main/java/com/google/cloud/datastore/Datastore.java]: This doesn't make sense to me, Can you please update this? 

May be something like `Reserve a list of IDs from a complete key` => `Reserve a list of IDs from a multiple keys`

File: google-cloud-datastore/src/main/java/com/google/cloud/datastore/Datastore.java [hunk 2]
```diff
@@ -168,6 +169,23 @@
...
+   *
+   * <p>Example of reserving multiple ids in a single batch.
+   *
+   * <pre>{@code
+   * KeyFactory keyFactory = datastore.newKeyFactory().setKind("MyKind");
+   * Key key1 = keyFactory.newKey(10);
+   * Key key2 = keyFactory.newKey("name");
+   * ReserveIdsResponse response = datastore.reserveIds(key1, key2);
+   *
```
Comment (rahulKQL) [inline google-cloud-datastore/src/main/java/com/google/cloud/datastore/Datastore.java:182]: this extra line can be deleted

**Review by rahulKQL — APPROVED** [review]
Other than JavaDoc nit LGTM

File: google-cloud-datastore/src/main/java/com/google/cloud/datastore/Datastore.java (hunk 2 above)
Comment (athakor) [reply google-cloud-datastore/src/main/java/com/google/cloud/datastore/Datastore.java:182]: yes we can, but i did it same as other javaDoc are there.

File: google-cloud-datastore/src/main/java/com/google/cloud/datastore/Datastore.java (hunk 1 above)
Comment (athakor) [reply google-cloud-datastore/src/main/java/com/google/cloud/datastore/Datastore.java]: done

**Pull request closed without merging**